### PR TITLE
chore(karpenter-core): go get github.com/aws/karpenter-core@v0.32.2-0.20231122223643-d0156e68fd30 (1/6)

### PR DESCRIPTION
**Pull request by charliedmcb**
Fixes # 

**Description**
We are trailing behind on core versions. This is one of many bumps to get updated on core.

Based off of this PR from aws doing the same version bump:
https://github.com/aws/karpenter-provider-aws/commit/f127d0ba74262ee001538c35325dd3cd3481bc6b

**How was this change tested?**
Nothing additional
*

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note
NONE
```

### CODE REVIEW

**Review by charliedmcb — COMMENTED** [review]
/test

File: go.mod [hunk 1]
```diff
@@ -18,7 +18,7 @@ require (
 	github.com/Azure/skewer v0.0.19
 	github.com/Pallinder/go-randomdata v1.2.0
 	github.com/alecthomas/units v0.0.0-20211218093645-b94a6e3cc137
-	github.com/aws/karpenter-core v0.32.3
+	github.com/aws/karpenter-core v0.32.2-0.20231122223643-d0156e68fd30
```
[the remark this answers is not in the document]
Comment (charliedmcb) [reply go.mod:21]: Note: this is only temp, as I have the PRs staged to transition to v0.33.0

File: pkg/cloudprovider/cloudprovider.go [hunk 2]
```diff
@@ -189,9 +188,13 @@ func (c *CloudProvider) Delete(ctx context.Context, nodeClaim *corev1beta1.NodeC
 
 func (c *CloudProvider) IsDrifted(ctx context.Context, nodeClaim *corev1beta1.NodeClaim) (cloudprovider.DriftReason, error) {
 	// Not needed when GetInstanceTypes removes nodepool dependency
-	nodePool, err := nodeclaimutil.Owner(ctx, c.kubeClient, nodeClaim)
-	if err != nil {
-		return "", client.IgnoreNotFound(fmt.Errorf("resolving owner, %w", err))
+	nodePoolName, ok := nodeClaim.Labels[corev1beta1.NodePoolLabelKey]
```
[the remark this answers is not in the document]
Comment (Bryce-Soghigian) [reply pkg/cloudprovider/cloudprovider.go:191]: What does AWS do?

[the remark this answers is not in the document]
Comment (tallaxes) [reply pkg/cloudprovider/cloudprovider.go:191]: @comtalyst's version should be fine for as long as drift detection does not need instance types (GetInstanceTypes needs nodepool - at least currently, see comment). Ours currently does not, AWS does (in isAMIDrifted).

**Review by tallaxes — APPROVED** [review]
LGTM, but consider IsDrifted simplification that @comtalyst suggested

File: pkg/cloudprovider/cloudprovider.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (charliedmcb) [reply pkg/cloudprovider/cloudprovider.go:191]: Bryce the following which matches my code here is what `aws` does
```
// Not needed when GetInstanceTypes removes nodepool dependency
	nodePoolName, ok := nodeClaim.Labels[corev1beta1.NodePoolLabelKey]
	if !ok {
		return "", nil
	}
	nodePool := &corev1beta1.NodePool{}
	if err := c.kubeClient.Get(ctx, types.NamespacedName{Name: nodePoolName}, nodePool); err != nil {
		return "", client.IgnoreNotFound(err)
	}
	if nodePool.Spec.Template.Spec.NodeClassRef == nil {
		return "", nil
	}
```

[the remark this answers is not in the document]
Comment (charliedmcb) [reply pkg/cloudprovider/cloudprovider.go:191]: Fair point about us not needing nodepool for now. Can always add it back later if needed.

**Review by Bryce-Soghigian — DISMISSED** [review]

**Review by charliedmcb — COMMENTED** [review]
/test

**Review by tallaxes — APPROVED** [review]

**Review by tallaxes — DISMISSED** [review]
Conditional on fixing (possibly removing?) the test

**Review by tallaxes — APPROVED** [review]

**Pull request merged**